vassal-app/src/main/java/VASSAL/counters/PieceDefiner.java

```diff
@@ -1281,15 +1282,47 @@ public Component getListCellRendererComponent(
       // It is incredibly inefficient for GamePieces and is not needed
       // since we overwrite the label text anyway.
       super.getListCellRendererComponent(list, "", index, selected, hasFocus);
+
       if (value instanceof EditablePiece) {
-        setText(((EditablePiece) value).getDescription());
+        final EditablePiece editableValue = (EditablePiece) value;
+        final String type = editableValue.getType().split(";")[0];
+        final String valueStr = editableValue.toString();
+        final String name = valueStr.substring(valueStr.indexOf("[name=")).split("=")[1].split(",")[0]; // extract name
+
+        // Prototype name field is null; skip first (dummy) item to avoid outputting a blank line
+        if (!name.isEmpty() || index > 0) {
```

Comment:
This doesn't work. A Basic Piece is allowed to have an empty name and the piece name is defined in a Basic Name trait. 

You will need to pass the PieceDefiner into the Renderer via a new constructor so the renderer can query the PieceDefiner if it editing a Propertype or not

In the PieceDefiner:

Public PieceDefiner() {
   ...
   r = new Renderer(this);
   ...
}


public boolean isPrototype() {
  return prototypeName.isEmpty()
}

In the Renderer:

private PieceDefiner definer;

Renderer(PieceDefiner definer) {
   this.definer = definer;
}

...

if definer.isPrototype() {
...

Reply:
Thanks. I couldn't work that out before. Next build incorporates these changes.